tianshou/data/batch.py

```diff
@@ -501,50 +545,97 @@ def to_torch(self, dtype: Optional[torch.dtype] = None,
             elif isinstance(v, Batch):
                 v.to_torch(dtype, device)
 
-    def cat_(self,
-             batches: Union['Batch', List[Union[dict, 'Batch']]]) -> None:
+    def __cat(self,
```

Comment:
- take advantage of the new code archiecture to make `lens` argument required and not optional
- move the computation of the initial value in `cat` itself. It will make the complete implementation clearer and simpler.

Reply:
Fixed in 6287326 .